src/main/java/com/chaos131/physics/Ballistics.java

```diff
@@ -0,0 +1,79 @@
+package com.chaos131.physics;
+
+import edu.wpi.first.math.Vector;
+import edu.wpi.first.math.numbers.N3;
+
+/**
+ * A utility class to calculate release vectors. Most functions are static and state-less.
+ * In another language, like C++, we would probably toss this in a namespace instead.
+ */
+public class Ballistics {
+    /**
+     * Calculates a release vector when trying to launch a game piece up into the air.
+     * It solves the equation ignoring air resistance, and uses the target_speed as the
+     * upwards speed at the moment the game piece reaches the target position. All points
+     * must be in a consistent coordinate frame. Either stick in field coordinates, or
+     * derive robot relative coordinates.
+     * 
+     * <p>Note that because this ignores air resistance, it is not a full physics model,
+     * but it should be accurate enough in most game scenarios if the mechanical team
+     * designed their system well.
+     * 
+     * @param release_position the coordinate of the release point, not the robot pose
+     * @param target_position the coordinate of the target point we want the ballistic to reach
+     * @param robot_velocity the speed of the robot at that instant
+     * @param target_speed the upwards speed of the game piece as it reaches the target_position
+     * @return the final release vector
+     */
+    public static Vector<N3> CalculateReleaseForUpwardsTarget(Vector<N3> release_position,
```

Comment:
I'm assuming this is code we will have some students fill in the gaps? We might want to add a note about that to the javadoc comment until it's implemented.

Reply:
Good idea!